vis_gen.py

```diff
@@ -99,63 +107,109 @@ def __init__(self, font_size=16, image_size=20,
         # Arg: grayscale
         self.grayscale = grayscale
 
+        # Set attribute _check_out_dir is a flag to avoid multiple checks on
+        # the existence of self.out_dir. For example, when function
+        # visualize_range() calls visualize_single() repeatedly, there is no
+        # need to repeated check the existence of out_dir in every call.
         self._check_out_dir = True
 
 
         
     @property
     def font_size(self):
+        """
+        Returns:
+             self._font_size: Int or Float, positive number indicating font
+                size.
+        """
         return self._font_size
     
     @property
     def image_size(self):
+        """
+        Returns:
+             self._image_size: Int or Float, positive number indicating image
+                size. This value is always larger than font_size.
+        """
         return self._image_size
 
     @property
     def font_name(self):
+        """
+        Returns:
+             self._font_name: Str, name of the font face.
+        """
         return self._font_name
 
     @property
     def font_style(self):
+        """
+        Returns:
+            self._font_style: Str, style of the font face ("Thin", "Bold",
+                "Regular"...). If not font_style set by user, returns empty
+                string "".
+        """
         return self._font_style
 
     @property
     def antialias(self):
+        """
+        Returns:
+             antialias: Str, one of "Default", "Best", "Fast", "Good", "None".
+        """
         return self._inv_antialias_map[self._antialias]
 
     @property
     def out_dir(self):
+        """
+        Returns:
+            self._out_dir: Str, relative path of the output directory.
+        """
         return self._out_dir
 
     @property
     def grayscale(self):
+        """
+        Returns:
+            self._grayscale: Bool, true if format of the output is in garyscale.
+        """
         return self._grayscale
 
 
     @font_size.setter
     def font_size(self, font_size):
         """Check if font_size is 
             1. larger than 0
-            2. smaller image_size"""
+            2. smaller image_size
```

Comment:
How about rewrite it as 
Raise ValueError if font_size is 0 or smaller or if font_size is larger than image_size.